packages/@react-stately/select/src/useSelectState.ts

```diff
@@ -62,25 +63,28 @@ export function useSelectState<T extends object>(props: SelectStateOptions<T>):
   });
 
   let [isFocused, setFocused] = useState(false);
+  let isEmpty = useMemo(() => listState.collection.size === 0 || (listState.collection.size === 1 && listState.collection.getItem(listState.collection.getFirstKey()!)?.type === 'loader'), [listState.collection]);
+  let open = useEffectEvent((focusStrategy: FocusStrategy | null = null) => {
```

Comment:
why?

Reply:
the main change here is to prevent the select from opening if it only has the loading sentinel in its collection. Otherwise it was just to make it a bit nicer but I can discard the effect event and revert it to define open/toggle in the return